packages/core/src/components/tree-view/tree-view.tsx

```diff
@@ -0,0 +1,214 @@
+   * the TreeView's collection (usually set the closed state on the last TreeViewItem in path).
+   */
+  @Event()
+  closeItem!: EventEmitter<TreeViewItem<string>[]>;
+
+  /**
+   * Emitted when a tree view item is clicked.
+   * The `detail` property of the `CustomEvent` will contain the complete path of TreeViewItems from the
+   * root to the item that is emitting the clicked event.
+   */
+  @Event()
+  clickItem!: EventEmitter<TreeViewItem<string>[]>;
+
+  keyDownListener = (event: KeyboardEvent) => {
+    if (event.defaultPrevented) {
+      return;
+    }
+
+    const isIndexLetter = (str: string) => str.length === 1 && str.match(/\S/);
+
+    const tree = event.composedPath().find(item => (item instanceof HTMLElement) ? item.className === 'dso-tree' : false);
+
+    if (!(event.target instanceof HTMLParagraphElement) || !(tree instanceof HTMLElement)) {
+      return;
+    }
+
+    switch (event.key) {
+      case "ArrowDown":
+        TreeView.moveFocus(tree, event.target, 'next');
+        break;
+      case "ArrowUp":
+        TreeView.moveFocus(tree, event.target, 'previous');
+        break;
+      case "ArrowRight":
+        TreeView.expandItemOrFocusChild(tree, event.target);
+        break;
+      case "ArrowLeft":
+        TreeView.collapseItemOrFocusParent(tree, event.target);
+        break;
+      case "End":
+        TreeView.moveFocus(tree, event.target, 'last');
+        break;
+      case "Home":
+        TreeView.moveFocus(tree, event.target, 'first');
+        break;
+      case "Enter":
+      case " ":
+        event.target.click();
+        break;
+      default:
+        if (isIndexLetter(event.key)) {
+          if (TreeView.setFocusByFirstCharacter(tree, event.target, event.key, event.shiftKey)) {
+            break;
+          }
+        }
+
+        return;
+    }
+
+    event.preventDefault();
+  };
+
+  itemClick = (event: Event, ancestors: TreeViewItem<string>[], item: TreeViewItem<string>) => {
+    if (!(event.target instanceof HTMLElement)) {
+      return;
+    }
+
+    if (event.target.classList.contains('tree-content') || event.target.parentElement?.classList.contains('tree-content')) {
+      this.clickItem.emit([...ancestors, item]);
+      return;
+    }
+
+    if (item.open) {
+      this.closeItem.emit([...ancestors, item]);
+    }
+    else {
+      this.openItem.emit([...ancestors, item]);
+    }
+  }
+
+  private static setFocus(tree: HTMLElement, target: HTMLElement) {
+    if (target) {
+      (Array.from(tree.querySelectorAll('p')) as HTMLElement[])
+        .filter(item => item.tabIndex === 0)
+        .forEach(item => item.tabIndex = -1);
+
+      target.tabIndex = 0;
+      target.focus();
+    }
+  }
+
+  private static moveFocus(tree: HTMLElement, el: HTMLElement, moveTo: 'first' | 'previous' | 'next' | 'last'): void {
+    const focusableItems = (Array.from(tree.querySelectorAll('p')) as HTMLElement[])
```

Comment:
Vanwaar de cast van `HTMLParagraphElement[]` naar `HTMLElement[]`?

Reply:
Als ik hier zou casten naar `HTMLParagraphElement[]` werkt `focusableItems.indexOf` niet meer goed denk ik. De parameter `el` is voldoende getypeerd op HTMLElement lijkt me. Of bedoel je juist "Waarom `el` niet als `HTMLParagraphElement` getypeerd?"